packages/stream-node/reader.js

```diff
@@ -0,0 +1,52 @@
+// @ts-check
+/// <reference types="ses"/>
+
+// This module provided for sake of fewer head scratches.
+// Node.js readable streams satisfy the signature of an async iterable iterator.
+// They however iterate Node.js Buffer values and are not hardened, so this
+// implementation compensates for both.
+
+const { details: X, quote: q } = assert;
+
+/**
+ * @param {import('stream').Readable} input the source Node.js reader
+ * @returns {import('@endo/stream').Reader<Uint8Array, void>}
+ */
+export const makeNodeReader = input => {
+  assert(
+    !input.readableObjectMode,
+    X`Cannot convert Node.js object mode Reader to AsyncIterator<Uint8Array>`,
+  );
+  assert(
+    input.readableEncoding === null,
+    X`Cannot convert Node.js Reader with readableEncoding ${q(
+      input.readableEncoding,
+    )} to a AsyncIterator<Uint8Array>`,
+  );
+  const iterator = input[Symbol.asyncIterator]();
+  const reader = harden({
+    async next() {
+      const result = await iterator.next();
+      if (result.done) {
+        return result;
+      }
+      assert(typeof result.value !== 'string');
+      return {
+        done: false,
+        value: new Uint8Array(result.value.buffer),
```

Comment:
Unfortunately that's not always right: https://nodejs.org/api/buffer.html#bufbyteoffset

I'm not sure what's the best way to do this while avoiding a copy.

Reply:
Addressed. I believe in this case, it might be just working out because the underlying stream has to make a copy from a ring buffer so it’s always aligned, but in principle it could be offset and it’s easy to accommodate the possibility.